RUFAS/biophysical/manure/manure_manager.py

```diff
@@ -0,0 +1,455 @@
+            raise ValueError(f"Unknown Processor Name found for {unknown_processor_names}.")
+
+    def _check_for_processors_without_connection_definition(
+            self,
+            all_referenced_processor_names: set[str],
+            processor_connections_by_name: dict[str, dict[str, list[dict[str, Any]]]]
+    ) -> None:
+        """
+        Checks for processors that are referenced but lack connection definitions.
+
+        Parameters
+        ----------
+        all_referenced_processor_names : set[str]
+            A set of names of all processors that are referenced and expected to have routing configurations.
+        processor_connections_by_name : dict[str, dict[str, list[dict[str, Any]]]]
+            A mapping of processor names to their routing connections, defining the configuration details.
+
+        Raises
+        ------
+        ValueError
+            If any processors are found to be missing a routing configuration.
+        """
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._check_for_processors_without_connection_definition.__name__,
+        }
+        processors_without_connection_definition: set[str] = set()
+        for processor_name in all_referenced_processor_names:
+            if processor_name not in processor_connections_by_name.keys():
+                processors_without_connection_definition.add(processor_name)
+                self._om.add_error(
+                    "Undefined Processor Connection.",
+                    f"No routing configurations found for {processor_name}.",
+                    info_map
+                )
+        if len(processors_without_connection_definition) > 0:
+            raise ValueError(f"Undefined Routing Connections for {processors_without_connection_definition}.")
+
+    def _find_all_referenced_processor_names(self, processor_connections: list[dict[str, Any]]) -> set[str]:
+        """
+        Retrieves all referenced processor names from a list of processor connections.
+
+        Parameters
+        ----------
+        processor_connections : list of dict
+            A list containing dictionaries that define connections between processors. Each dictionary is expected to
+            have a "processor_name" key, and either "solid_output_destinations" and "liquid_output_destinations",
+            or "destinations".
+
+        Returns
+        -------
+        set[str]
+            A set of all unique processor names (both as origin and as destination) referenced in the connections.
+        """
+        all_referenced_processor_names: set[str] = set()
+        for origin in processor_connections:
+            origin_processor_name = origin["processor_name"]
+
+            is_separator: bool = ("solid_output_destinations" in origin.keys()
+                                  and "liquid_output_destinations" in origin.keys())
+            all_referenced_processor_names.add(origin_processor_name)
```

Comment:
```suggestion
        for origin in processor_connections:
            origin_processor_name = origin["processor_name"]
            all_referenced_processor_names.add(origin_processor_name)
            is_separator: bool = ("solid_output_destinations" in origin.keys()
                                  and "liquid_output_destinations" in origin.keys())
```
Just seems a little clearer rather than checking for the whether it's a separator between extracting the name and adding it to the list.